models/src/test/scala/coop/rchain/models/RhoTypesTest.scala

```diff
@@ -0,0 +1,39 @@
+package coop.rchain.models
+
+import org.scalacheck.Gen
+import org.scalatest.{Matchers, PropSpec}
+import org.scalatest.prop.PropertyChecks
+
+class RhoTypesTest extends PropSpec with PropertyChecks with Matchers {
+  def rtt[T](msg: T, serializeInstance: Serialize[T]): Boolean = {
+    val bytes = serializeInstance.encode(msg)
+    val msg2  = serializeInstance.decode(bytes)
+    assert(msg2.isRight && msg == msg2.right.get)
+
+    val errBytes = Array[Byte](1, 2, 3, 4)
+    val msg3     = serializeInstance.decode(errBytes)
+    assert(msg3.isLeft && msg3.left.get.isInstanceOf[SerializeError])
```

Comment:
No need for `asInstanceOf` after the above change.  This is sufficient:
```scala
assert(msg3.isLeft)
```

Reply:
Accordingly to decode signature it may return any Throwable on error. Until we decide to change decode signature to Either[SerializeError, T] to force all derived classes (like stringInstance for example), wrap errors in SerializeError, I think it is better to check exact error type to catch unexpected serialization failures that may appear after a refactoring or other changes.

Reply:
Done.